samples/web/content/apprtc/apprtc.py

```diff
@@ -310,16 +311,38 @@ def __str__(self):
     return '{%r, %d}' % (self.is_initiator, len(self.messages))
 
 class Room:
-  def __init__(self):
+  # Room used for simple named rooms, can be joined by knowing the room name.
+  TYPE_OPEN = 1
+  # Room used for direct calling, can only be joined by allowed clients.
+  TYPE_DIRECT = 2
+  
+  def __init__(self, room_type):
     self.clients = {}
+    # The list of allowed clients will include the initiator and other
+    # clients allowed to join the room. If no clients are added, no
+    # list is enforced, and is_client_allowed will always return true.
+    self.allowed_clients = None;
+    if room_type != Room.TYPE_OPEN and room_type != Room.TYPE_DIRECT:
+      room_type = Room.TYPE_OPEN
+    self.room_type = room_type
   def add_client(self, client_id, client):
     self.clients[client_id] = client
+  def add_allowed_client(self, client_id):
+    if self.allowed_clients is None:
+      self.allowed_clients = []
+    if client_id not in self.allowed_clients:
+      self.allowed_clients.append(client_id)
   def remove_client(self, client_id):
     del self.clients[client_id]
   def get_occupancy(self):
     return len(self.clients)
   def has_client(self, client_id):
     return client_id in self.clients
+  def is_client_allowed(self, client_id):
+    # If allowed_clients is None, don't enforce allowed_client list.
+    if self.allowed_clients is None:
```

Comment:
nit: can do:
return self.allowed_clients is None or client_id in self.allowed_clients